Test på håndtering av POSIXct-variabler i valideringsdatasett

cahuse opened the pull request.

cahuse committed:
Test på håndtering av POSIXct-variabler i valideringsdatasett

cahuse committed:
Håndterer variabler med subklasser (flere klasser), eks. POSIXct-var

cahuse committed:
Valideringsdatasett med variabelnavn med _ skal regnes som gyldige

haugedal commented on R/lag_valideringsdatasett.R:
```diff
@@ -73,37 +73,40 @@ lag_valideringsdatasett = function(d_reg, indvars) {
 
 
   # Lag valideringsdatasett
-
```
Dette linjeskiftet skal vera her. Det viser at kommentaren gjeld fleire av kodeblokkene under.

cahuse replied:
Redigert.

haugedal commented on R/lag_valideringsdatasett.R:
```diff
@@ -73,37 +73,40 @@ lag_valideringsdatasett = function(d_reg, indvars) {
     left_join(d_kopling, by = "vld_varnamn", relationship = "many-to-one")
   antal_rader = nrow(d_vld)
   d_vld$rekkjefolgje = seq_len(antal_rader)
-
```
Dette linjeskiftet skal vera her.

cahuse replied:
Redigert.

haugedal commented on R/lag_valideringsdatasett.R:
```diff
@@ -73,37 +73,40 @@ lag_valideringsdatasett = function(d_reg, indvars) {
+      class(d_vld[[vld_varnamn]]) = unike_vartypar[[i]]
     }
   }
-
```
Dette linjeskiftet skal vera her.

cahuse replied:
Redigert.

haugedal commented on R/lag_valideringsdatasett.R:
```diff
@@ -122,10 +125,10 @@ lag_valideringsdatasett = function(d_reg, indvars) {
       purrr::map_df(flytt_resultat) %>%
       ungroup()
   }
-
```
Dette linjeskiftet skal vera her.

cahuse replied:
Redigert.

haugedal commented on R/lag_valideringsdatasett.R:
```diff
@@ -122,10 +125,10 @@ lag_valideringsdatasett = function(d_reg, indvars) {
     select(-all_of(datavars), -c("res_kol", "rekkjefolgje"))
   d_vld
 }
+
```
Dette linjeskiftet skal *ikkje* vera her, vil ikkje ha to linjeskift på slutten av fila.

cahuse replied:
Redigert.

haugedal commented on R/lag_valideringsdatasett.R:
```diff
@@ -73,37 +73,40 @@ lag_valideringsdatasett = function(d_reg, indvars) {
 
+  # Slå sammen elementer fra samme klass i en string
+  vartypar_sammenlagt = c()
+  for( i in seq_len(length(vartypar))){
```
Dette kan gjerast med `map_chr()`.

cahuse replied:
Redigert.

haugedal commented on R/lag_valideringsdatasett.R:
```diff
@@ -73,37 +73,40 @@ lag_valideringsdatasett = function(d_reg, indvars) {
   unike_vartypar = unique(vartypar)
 
+  # Slå sammen elementer fra samme klass i en string
+  vartypar_sammenlagt = c()
```
All info om vartypar bør lagrast i ei dataramme for å sikra parallellitet.

haugedal commented on R/lag_valideringsdatasett.R:
```diff
@@ -73,37 +73,40 @@ lag_valideringsdatasett = function(d_reg, indvars) {
     unname()
   unike_vartypar = unique(vartypar)
 
+  # Slå sammen elementer fra samme klass i en string
```
Kanskje `# Slå sammen klasser fra samme variabeltype til én string` er betre?

cahuse replied:
Redigert

cahuse replied:
Redigert.

haugedal reviewed (changes requested):
Sjå spesifikke kommentarar inne i filene.

Me oppdaga eit nytt problem, som vil oppstå dersom to ulike variablar er av typane `c("foo", "bar")` og `"foo_bar"`. Eg legg til ein test på dette, og så bør funksjonen stoppa med ei eksplisitt feilmelding i slike tilfelle.

Fleire plassar fylgjer ikkje koden kodestilguiden vår. Dette fiksar du enklast med å kjøra `styler` med `rapwhale_style()`.

haugedal committed:
Test på at funksjonen stoppar når samanslåing av klassar ikkje gjev eintydige resultat

cahuse committed:
Fiksert problem med variabeltype med ulike klasser

Det kunne hende at to ulike variabler med ulike klasser kunne ha samme
sammenlagt navn. Feil melding ble gjort for å melde om at lag_valideringsdatasett()
ikke kan brukes om sånne tilfeller.

haugedal commented on R/lag_valideringsdatasett.R:
```diff
@@ -74,39 +74,47 @@ lag_valideringsdatasett = function(d_reg, indvars) {
+
+  d_vartyper = tibble::tibble(vartypar = vartypar, 
+                              vartypar_sammenlagt = vartypar_sammenlagt
+                              )
```
Dataramma bør vel lagast direkte, og ikkje ved å fyrst laga to vektorar? Poenget var jo å sikra parallellitet, og at ikkje nokon seinare endrar rekkjefylgja i ein av vektorane, utan å veta at koden nedanfor er avhenging av henne.

cahuse replied:
Har nå definert  variablene inn i dataramme.

haugedal commented:
Det er framleis kode (blant anna nokre innrykk) som ikkje fylgjer kodestilguiden vår (køyr `styler` med `rapwhale_style()`).

cahuse committed:
Definert d_vartypar og d_vartypar_sammenlagt direkt i dataramme

Målet med dette er å sikra parallellitet, og at ikkje nokon
seinare endrar rekkjefylgja i ein av vektorane, utan å veta
 at koden nedanfor er avhenging av henne

huftis commented on R/lag_valideringsdatasett.R:
```diff
@@ -71,15 +71,28 @@ lag_valideringsdatasett = function(d_reg, indvars) {
+    vartypar = map(d_reg[datavars], ~ class(.x)) %>%
+      unname(), # Lag oversikt over variabeltypar
+
+    vartypar_sammenlagt = map_chr(
```
Foreslår heller slike linjeskift:
```r
vartypar_sammenlagt = map_chr(
  vartypar,
  \(x) paste(x, collapse = "_")
)
```

cahuse replied:
Redigert.

cahuse replied:
Redigert.

huftis commented on R/lag_valideringsdatasett.R:
```diff
@@ -71,15 +71,28 @@ lag_valideringsdatasett = function(d_reg, indvars) {
+      \(x) paste(x,
+        collapse = "_"
+      )
+    ) # Slå sammen elementer fra samme klass i en string
```
Ev. `# Slå sammen alle (under)klassenavn til én tekststreng`.
Men eigentleg er det vel ikkje nødvendig med ein kommentar i det heile tatt?

cahuse replied:
Gjerne la den stå for tiden.

huftis replied:
OK.

haugedal commented on tests/testthat/test-lag_valideringsdatasett.R:
Denne fila fylgjer framleis ikkje kodestilen (køyr styler).

cahuse replied:
Gjort

cahuse committed:
Kjørt styler med rapwhale_style()

cahuse committed:
Byttet linjeskifte og fjernet kommentar som foreslått

haugedal commented on R/lag_valideringsdatasett.R:
Hugs å importer funksjonar brukte frå andre pakkar med `@importFrom` i toppen av fila (`tibble()`, `map()`, `map_chr()`, `distinct()`...).

cahuse replied:
Gjort.

cahuse replied:
Gjort.

haugedal replied:
`@importFrom` skal stå i starten av fila, etterfulgt av ei linje med `NULL`. Sjå eksemel i andre filer i `rapwhale`.

Og `setdiff()` kjem vel ikkje frå `lubridate`. Han finst i `base` (og `generics` og `dplyr`), men er vel unødvendig å importera her.

cahuse replied:
Gjort.

cahuse committed:
Lagt inn @importFrom til funksjoner som skal importeres

cahuse commented:
> Sjå spesifikke kommentarar inne i filene.
> 
> Me oppdaga eit nytt problem, som vil oppstå dersom to ulike variablar er av typane `c("foo", "bar")` og `"foo_bar"`. Eg legg til ein test på dette, og så bør funksjonen stoppa med ei eksplisitt feilmelding i slike tilfelle.
> 
> Fleire plassar fylgjer ikkje koden kodestilguiden vår. Dette fiksar du enklast med å kjøra `styler` med `rapwhale_style()`.

Disse er gjort nå, men får ikke "Resolved" knappen

cahuse committed:
Satt @import med funksjoner på toppen av skriptet

haugedal committed:
Fjerna unødvendige linjeskift og mellomrom
